test/test_helpers.jl

```diff
@@ -0,0 +1,24 @@
+using LinearAlgebra
+using SparseArrays
+
+"""
+    equivalent_matrices(x::AbstractMatrix)
+
+Returns a collection of matrixes that are equal to the input `x`, but of a different type.
+Useful for testing if things still work on different types of matrix.
+"""
+function equivalent_matrices(x::AbstractMatrix)
```

Comment:
If we wanted to go more intense with this we could nest if so we would see things like `view(sparse(x), :, :)`
etc.